packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -759,7 +759,7 @@ class ListTile extends StatelessWidget {
       ?? MaterialStateMouseCursor.clickable.resolve(states);
 
     return InkWell(
-      customBorder: shape ?? tileTheme.shape,
+      customBorder: shape ?? tileTheme.shape ?? theme.listTileTheme.shape,
```

Comment:
This would make the `ListTileTheme` handle the outer `ThemeData` settings differently than other widgets. The overall `ThemeData.listTileTheme` is already taken into account in `ListTileThemeData.of`.

The problem here appears to be the fact that `Drawer` wraps its child in a `ListThemeData` that overrides the text style:

https://github.com/flutter/flutter/blob/332c22f3912be26e6adf9c0908f6d469495e6010/packages/flutter/lib/src/material/drawer.dart#L667-L670

Which causes the outer ThemeData entry to be ignored for things like shape when inside a Drawer. 

To be more consistent with the rest of the way themes are handled, I think a better way to handle this would be to have the Drawer wrap its child with `ListThemeData.merge` so that it only overrides the style parameter of whatever the current surrounding theme is.

Reply:
That makes sense, thank you for this feedback!